src/hpr_config.erl

```diff
@@ -0,0 +1,178 @@
+-module(hpr_config).
+
+-export([
+    init/0,
+    stop/0,
+    update_routes/1,
+    insert_route/1,
+    lookup_devaddr/1,
+    lookup_eui/2
+]).
+
+-define(DEVADDRS_ETS, hpr_config_routes_by_devaddr).
+-define(EUIS_ETS, hpr_config_routes_by_eui).
+
+-spec init() -> ok.
+init() ->
+    ?DEVADDRS_ETS = ets:new(?DEVADDRS_ETS, [public, named_table, bag, {read_concurrency, true}]),
+    ?EUIS_ETS = ets:new(?EUIS_ETS, [public, named_table, bag, {read_concurrency, true}]),
+    ok.
+
+-spec stop() -> ok.
+stop() ->
+    true = ets:delete(?DEVADDRS_ETS),
+    true = ets:delete(?EUIS_ETS),
+    ok.
+
+-spec update_routes(map()) -> ok.
+update_routes(#{routes := Routes}) ->
+    {NewDevaddrRows, NewEUIRows} = lists:foldl(
+        fun(RouteConfig, {DevaddrRowsAcc, EUIRowsAcc}) ->
+            Route = hpr_route:new(RouteConfig),
+            DevaddrRowsAcc2 = DevaddrRowsAcc ++ route_to_devaddr_rows(Route),
```

Comment:
Do we expect the list of routes to be long? It might be more efficient to build up a list of lists and then use `lists:append` to flatten after the `foldl`.

Reply:
I think it should be few hundred items probably

Reply:
My concern is that `++` here is O(n^2)ish, so if we expect that there are a lot of routes, we should avoid `++`. Alternatively, we could insert to `ets` along the way, rather than accumulating a large list to insert all at once.

Reply:
fair enough 👍🏻